packages/app/src/app/pages/Sandbox/Editor/Workspace/items/Deployment/Netlify.tsx

```diff
@@ -0,0 +1,200 @@
+import { Button } from '@codesandbox/common/lib/components/Button';
+import { resolveDirectory } from '@codesandbox/common/lib/sandbox/modules';
+import getTemplate from '@codesandbox/common/lib/templates';
+import { observer } from 'mobx-react-lite';
+import React, { useEffect, useState } from 'react';
+import LinkIcon from 'react-icons/lib/fa/external-link';
+import Cogs from 'react-icons/lib/fa/cogs';
+import LightningIcon from 'react-icons/lib/md/flash-on';
+
+import DeploymentIntegration from 'app/components/DeploymentIntegration';
+import NetlifyLogo from 'app/components/NetlifyLogo';
+import { useSignals, useStore } from 'app/store';
+import getNetlifyConfig from 'app/utils/getNetlifyConfig';
+
+import { WorkspaceInputContainer, WorkspaceSubtitle } from '../../elements';
+
+import {
+  Deploys,
+  Deploy,
+  Name,
+  Link,
+  DeploysWrapper,
+  Wrapper,
+  ButtonContainer,
+} from './elements';
+
+const getFunctionDir = sandbox => {
+  try {
+    return resolveDirectory(
+      getNetlifyConfig(sandbox).functions,
+      sandbox.modules,
+      sandbox.directories
+    );
+  } catch (e) {
+    return [];
+  }
+};
+
+export const Netlify = observer(() => {
+  const {
+    deployment: { getNetlifyDeploys, deployWithNetlify },
+    modalOpened,
+  } = useSignals();
+  const { deployment, editor } = useStore();
```

Comment:
@Saeris I think you should destructur `deployment` & `editor`  just like you did in the other files

```suggestion
  const {
    deployment: { building, deploying, netlifyClaimUrl, netlifyLogs, netlifySite },
    editor: { currentSandbox },
  } = useStore();
```

Reply:
Hmm, for me it looks cleaner to not destructure everything, makes it harder to read.

Reply:
It's  because @Saeris did it in all his other commits too, so to keep the code consistent.

Reply:
I found during development of these refactors that sometimes errors are thrown when you overly destructure like this, that's why it's inconsistent where I drill down this deeply. Right now I'm not sure what the rules should be around destructuring.

Personally I think it's good to declare all of the state properties you'll be using in a component up at the top because you can see at a glance what a component depends on to render. I'll have to talk to Ives on this one.